### PR TITLE
feat: PutComponentMetric access control policy validation

### PR DESCRIPTION
**Pull request by kucnavya**
**Issue #, if available:**

**Description of changes:**
Add resource validation to authorization checks in PutComponentMetric handler.
Add ComponentMetricIPCService to kernel lifecycle. 

**Why is this change necessary:**

**How was this change tested:**
- [x] Updated or added new unit tests.
- [ ] Updated or added new integration tests.
- [ ] Updated or added new end-to-end tests.
- [ ] If my code makes a remote network call, it was tested with a proxy.

**Any additional information or context required to review the change:**

**Documentation Checklist:**
 - [ ] Updated the README if applicable.

**Compatibility Checklist:**
- [ ] I confirm that the change is backwards compatible.
- [ ] Any modification or deletion of public interfaces does not impact other plugin components.
- [ ] For external library version updates, I have reviewed its change logs and Nucleus does not consume 
  any deprecated method or type.

Refer to [Compatibility Guidelines](/COMPATIBILITY.md) for more information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by nikkhilmuthye — DISMISSED** [review]

**Review by rightpeter — DISMISSED** [review]

File: src/test/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgentTest.java [hunk 1]
```diff
@@ -116,6 +116,20 @@ void GIVEN_put_component_metric_request_with_invalid_service_name_WHEN_handle_re
         }
     }
 
+    @Test
+    void GIVEN_put_component_metric_request_with_invalid_metric_name_WHEN_handle_request_called_THEN_throw_exception()
```
[the remark this answers is not in the document]
Comment (kucnavya) [reply src/test/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgentTest.java]: done.

File: src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java [hunk 2]
```diff
@@ -167,28 +172,44 @@ private TelemetryUnit valueOfIgnoreCase(String unitAsString) {
...
+    private void doMetricAuthorization(String opName, String serviceName, String metricName)
+            throws AuthorizationException {
         if (authorizationHandler.isAuthorized(PUT_COMPONENT_METRIC_SERVICE_NAME,
-                Permission.builder().operation(opName).principal(serviceName).resource("*").build())
-                && SERVICE_NAME_REGEX.matcher(serviceName).find()) {
+                Permission.builder().operation(opName).principal(serviceName).resource(metricName).build())) {
+            return;
+        }
+        throw new AuthorizationException(
+                String.format("Principal %s is not authorized to perform %s:%s with metric " + "name %s", serviceName,
```
[the remark this answers is not in the document]
Comment (kucnavya) [reply src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java]: done

**Review by shaguptashaikh — APPROVED** [review]

File: src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java [hunk 3]
```diff
@@ -167,28 +172,44 @@ private TelemetryUnit valueOfIgnoreCase(String unitAsString) {
         return telemetryEnum;
     }
 
-    // Validate metric request
-    // Check name, unit and value arguments
-    private void validateComponentMetricRequest(List<software.amazon.awssdk.aws.greengrass.model.Metric> metrics) {
+    // Validate metric request - check name, unit and value arguments
+    // Also authorize request against access control policy
+    // throw IllegalArgumentException is request params are invalid
```
Comment (J-C-L) [inline src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java:177]: nit: typo "if request params are invalid"

File: src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java [hunk 4]
```diff
@@ -167,28 +172,44 @@ private TelemetryUnit valueOfIgnoreCase(String unitAsString) {
...
                         String.format("Invalid argument found in PutComponentMetricRequest"));
             }
+            doMetricAuthorization(opName, serviceName, metric.getName());
         }
     }
 
     // Validate if request is authorized for given operation
-    // Also validate if serviceName is of format "aws.*"
-    private void doAuthorization(String opName, String serviceName) throws AuthorizationException {
+    // throws AuthorizationException is not authorized
```
Comment (J-C-L) [inline src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java:197]: nit: typo "throws AuthorizationException if not authorized"

**Review by J-C-L — APPROVED** [review]

**Review by shaguptashaikh — APPROVED** [review]

**Review by nikkhilmuthye — APPROVED** [review]

**Pull request merged**